wallet-new/test/unit/Test/Spec/Keystore.hs

```diff
@@ -0,0 +1,111 @@
+{-# OPTIONS_GHC -fno-warn-orphans       #-}
+{-# LANGUAGE ViewPatterns #-}
+module Test.Spec.Keystore (
+    spec
+  ) where
+
+import           Universum
+
+import           System.Directory (doesFileExist, removeFile)
+import           System.IO.Error (IOError)
+
+import           Test.Hspec (Spec, describe, it, shouldBe, shouldReturn)
+import           Test.Hspec.QuickCheck (prop)
+import           Test.QuickCheck (Gen, arbitrary)
+import           Test.QuickCheck.Monadic (forAllM, monadicIO, pick, run)
+
+import           Pos.Crypto (EncryptedSecretKey, hash, safeKeyGen)
+
+import           Cardano.Wallet.Kernel.DB.HdWallet (eskToHdRootId)
+import           Cardano.Wallet.Kernel.Keystore (Keystore)
+import qualified Cardano.Wallet.Kernel.Keystore as Keystore
+import           Cardano.Wallet.Kernel.Types (WalletId (..))
+
+import           Util.Buildable (ShowThroughBuild (..))
+
+{-# ANN module ("HLint: ignore Reduce duplication" :: Text) #-}
+
+-- | Creates, operate on a keystore and finally destroys it.
+withKeystore :: (MonadIO m, MonadMask m) => (Keystore -> m a) -> m a
+withKeystore action =
+    bracket (Keystore.newInMemoryTransientKeystore)
+            (\ks -> Keystore.releaseKeystore ks >>
+                    Keystore.destroyKeystore ks
```

Comment:
Destroy also releases, so can just call `destroyKeystore`. Might make sense to call it `releaseAndDestroy`?